rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+ * Ongoing vulnerability scanning with Trivy
+
+#### **CIS vs STIG**
+
+The two best-known standards for security hardening are 
+ * CIS Benchmarks (Level 1 & 2), and
+ * The US DoD Security Technical Implementation Guides (STIGs)
+
+Both of these standards are pretty comprehensive and have wide
+application, although private organizations tend to favor CIS
+over STIG. 
+
+So why choose STIG over CIS?
+
+The decision largely comes down to what tooling is immediately available. Both
+standards are comprehensive - the AL2 CIS benchmark document runs to about 600 
+pages. At an estimated 2 pages per item, that's 300 separate items that need
+detection, remediation and validation. I don't believe that Teleport wants 
+to maintain scripts to do that ourselves, so I have been focused on finding 
+tools to do it for us.
+
+Somewhat unsurprisingly, nearly all of the automated hardening tools I have 
+found are tied to the AWS EC2 Image Builder Tool. Both standards are represented, 
+but while the STIG-compliance components are free for use on any image, the 
+CIS tooling has several drawbacks:
+
+  1. The hardening components require the use of a CIS-supplied base image that
+     incurs an added per-minute royalty to CIS. As we are using this image as a
+     delivery channel to our customers, this is a non-starter.
+  2. The use of the CIS hardening tools requires a subscription to/partnership with 
+     CIS that we do not currently have
+
+The decision to go with STIG is based on being able to improve the situation
+for our customers *now*, while still being able to add CIS hardening tools as
+we develop that relationship in the future.
+
+#### **Hardened Base Image**
+
+My proposed solution is to use AWS EC2 Image builder, using the available STIG
+tooling, to create a hardened base image. 
+
+We can do this as part of the build process, or on some frequent schedule 
+(e.g. daily, weekly).
+
+I suggest the latter, scheduled option;
+
+ 1. The release build process is already complicated enough, and
+ 2. There is no real advantage to creating a per-release base image, as there 
+    is no way to inject parameters into the image build pipeline.
+
+#### **Why ImageBuilder *AND* Packer?**
+
+As mentioned above, there is no way to inject parameters into an EC2 ImageBuilder
+pipeline. The pipeline is the pipeline. There is some customization available at
+the build component level, but there is no way (that I can see, at least) to
+say something like "Trigger this pipeline using Teleport `vX.Y.Z` and save the
+resulting image as `teleport-oss-X.Y.Z`". All you can do is trigger the pipeline
+and have it produce (or replace) a single image.
+
+So this brings us to an obvious division of labor:
+
+ * **EC2 ImageBuilder** to build the hardened base image, which will be used as
+   the common base for all AMIs we produce, across all supported Teleport
+   versions, and
+ * **Packer** for the final, individual AMIs that are tailored to a specific
```

Comment:
Won't such flow allow the Packer to reintroduce insecure configurations into supposedly hardened image?